gammapy/image/models/new.py

```diff
@@ -58,25 +60,96 @@ class SkyGaussian2D(SkySpatialModel):
 
     Parameters
     ----------
-    lon_mean : `~astropy.coordiantes.Longitude`
-        :math:`x_0`
-    lat_mean : `~astropy.coordinates.Latitude`
-        :math:`y_0`
+    lon_0 : `~astropy.coordiantes.Longitude`
+        :math:`lon_0`
+    lat_0 : `~astropy.coordinates.Latitude`
+        :math:`lat_0`
     sigma : `~astropy.coordinates.Angle`
         :math:`\sigma`
     """
 
-    def __init__(self, lon_mean, lat_mean, sigma):
+    def __init__(self, lon_0, lat_0, sigma):
         self.parameters = ParameterList([
-            Parameter('lon_mean', Longitude(lon_mean)),
-            Parameter('lat_mean', Latitude(lat_mean)),
+            Parameter('lon_0', Longitude(lon_0)),
+            Parameter('lat_0', Latitude(lat_0)),
             Parameter('sigma', Angle(sigma))
         ])
 
     @staticmethod
-    def evaluate(lon, lat, lon_mean, lat_mean, sigma):
+    def evaluate(lon, lat, lon_0, lat_0, sigma):
         """Evaluate the model (static function)."""
-        sep = angular_separation(lon, lat, lon_mean, lat_mean)
+        sep = angular_separation(lon, lat, lon_0, lat_0)
         fact = (sep / sigma).to('').value
-        val = np.exp(-0.5 * fact **2) / (2 * np.pi * sigma**2)
+        val = np.exp(-0.5 * fact ** 2) / (2 * np.pi * sigma**2)
         return val
+
+
+class SkyPointSource(SkySpatialModel):
+    r"""Point Source.
+
+    .. math::
+
+        \phi(x, y) = \delta{(lon - lon_0, lat - lat_0)}
+
+    A tolerance of 1 arcsecond is accepted for numerical stability
+
+    Parameters
+    ----------
+    lon_0: `~astropy.coordiantes.Longitude`
+        : math: `lon_0`
+    lat_0: `~astropy.coordinates.Latitude`
+        : math: `lat_0`
+    """
+
+    def __init__(self, lon_0, lat_0):
+        self.parameters = ParameterList([
+            Parameter('lon_0', Longitude(lon_0)),
+            Parameter('lat_0', Latitude(lat_0))
+        ])
+
+    @staticmethod
+    def evaluate(lon, lat, lon_0, lat_0):
+        """Evaluate the model (static function)."""
+        tolerance = 1 * u.arcsec
+        sep = angular_separation(lon, lat, lon_0, lat_0)
+        val = 1 if sep < tolerance else 0
```

Comment:
I discussed this with @cdeil and he suggested to leave the proper treatment (whatever that means) of the delta function to the function that evaluated the spatial models on the skycube. I guess the current implementation of ``SpatialModel`` is meant to be evaluated at a given coordinate, so everything involving pixel intergration schemes is out of scope. Do you agree?

Reply:
Yes, the point source needs to be special-cases in the model evaluator. It requires knowledge of the pixels (WCS or HPX), and probably it's not possible or at least not a good idea to attempt to do it here, i.e. couple this in the model code.

So this spatial model is mostly there to do book-keeping of the parameters.
I'm not sure yet what this measn for the evaluate method.
Leave as-is? Remove possible? Always return 1?
Maybe postpone this discussion until tomorrow, when we have a first draft map model evaluator and see more clearly how it could work for this case?

Reply:
When it is special cased in the model evaluator anyway, I would either do the 0th order approximation and just set the nearest pixel center to one or raise a `NotImplementedError`.

Reply:
Suggest to just leave as-is, and then continue the discussion tomorrow when we handle this in the model evaluator.